validator/client/validation_client.go

```diff
@@ -213,6 +228,16 @@ func (r *ExecutionClientRun) GetStepAt(pos uint64) containers.PromiseInterface[*
 }
 
 func (r *ExecutionClientRun) GetLeavesWithStepSize(fromBatch, machineStartIndex, stepSize, numDesiredLeaves uint64) containers.PromiseInterface[[]common.Hash] {
+	if r.client.boldValClient != nil {
+		return r.client.boldValClient.GetLeavesWithStepSize(&server_api.GetLeavesWithStepSizeInput{
```

Comment:
it works as an intermediate solution - but I'd like to see a separate BOLD-used interface for bold execution, and be able to see what parts of the interface are indeed needed in BOLD. This might be more of a BOLD comment than a comment for this PR.

Reply:
Acknowledged.